CONTRIBUTING.md

```diff
@@ -0,0 +1,114 @@
+    - Module/tool/makefile/script in question.
+    - Desired behavior / rationale for change.
+    - Proposed solution(s), including scope of work and affected modules.
+    - Methods of quantitative evaluation.
+- Example: Support pre-fetching in I$.
+### Informational
+**I don't understand this.**
+- Suggested fields:
+    - Descriptive title.
+    - Module/tool/makefile/script in question.
+    - Information requested.
+    - Background information already gathered.
+- Example: What algorithm does D$ LRU use?
+- Resolution: Additional comments, additional documentation.
+
+## Making a PR
+The absolute most important factor in making a successful open-source contribution is communication with the maintainers. Therefore it is highly recommended to raise an issue and discuss potential implementation paths before making a pull request. "Dumping code" is discouraged. 
+
+An ideal collaboration looks like the following:
+1. I would like BlackParrot to be able to eat ants.
+2. Open an issue
+    - Title: "Support for eating ants"
+    - Supporting eating ants would increase PPA because BlackParrot would have more energy.
+    - We could either modify the beak or the talons, with the tradeoff being that the beak is more efficent, but harder to modify.
+3. Communicate with BlackParrot maintainers
+    - We have an extensive talon modification pending on another branch, which will be incompatible with your proposal. Your beak proposal interesting though, have you considered the effects on eating berries?
+    - \<Back and forth\>
+    - Design specification document is written and agreed upon.
+4. Make a PR with an initial version of code.
+5. Iterate with BlackParrot maintainers on code quality, required testing, PPA analysis, etc.
+6. PR is merged! (°<°)
+
+## Tips for having your PR approved
+- Follow the [BlackParrot style guide](STYLE_GUIDE.md).
+- Make separate commits for moving files, cleaning files and functional changes.
+- Rebase away 'temporary' or 'broken' commits.
+- Code should lint in both Verilator and VCS (if you have access to it).
+- Don't commit intermediate or generated files.
+
+## Permanent branches
+- master: release branch
+    - Clients should be able to git clone and run a working copy at all times.
+    - Functionality, power, performance CI MUST pass before merge is accepted
+- dev: development branch  
+    - Developers should be able to git clone and run a buggy copy at all times.
+    - CI should pass before merge is accepted.  
+    - Occasional churn at this level is acceptable as there may be inter-module dependencies being worked on.
+- top_dev: bp_top development branch
+    - Top level wrapping modules; tiles, off-chip networks, SoC configurations.
+    - CI should often pass, but there is no gatekeeper at this level.  
+- fe_dev, be_dev, me_dev: Module level development branch
+    - All \<end\> work should branch off of and merge into this branch. fe_dev will periodically merge from and into dev.
```

Comment:
"fe_dev, be_dev, and me_dev will periodically merge from and into dev."

Not just fe_dev, right?

Reply:
fixed